tests/conftest.py

```diff
@@ -22,3 +25,9 @@ def invoke_with_config(self, *args, **kwargs):
             ["--config-file", self.test_snowcli_config, *args[0]],
             **kwargs,
         )
+
+
+@pytest.fixture(autouse=True)
+def reset_global_context_after_each_test(request):
+    snow_cli_global_context.reset_global_context()
```

Comment:
Personally I would consider doing here `snow_cli_global_context_manager.update_global_context(modifications)` instead of introducing test-only helper in production code. WDYT?

Reply:
A) I understand the argument about test-only helper in production code, but your approach would require knowing what are default values of context fields in test code and I don't think it's a good idea. Of course I could make `_create_snow_cli_global_context_manager_with_default_values` a public function but then it's the same problem - making a function public only for test purposes. And then this test-only helper is less verbose.

B) I tried a solution with module reload but then a lot of tests start failing because they import the module, use some objects from that module and then module reloading creates new instances different than these which are already imported and used in test code.

---

I'd stay with my solution.